morpheus/_lib/include/morpheus/utilities/tensor_util.hpp

```diff
@@ -111,13 +112,15 @@ struct MORPHEUS_EXPORT TensorUtils
      * @brief Compute the number of elements in a tensor based on the shape
      *
      * @tparam IndexT
+     * @tparam RetTypeT
      * @param shape
      * @return IndexT
      */
-    template <typename IndexT>
-    static inline IndexT get_elem_count(const std::vector<IndexT>& shape)
+    template <typename IndexT, typename RetTypeT = std::size_t>
+    static inline RetTypeT get_elem_count(const std::vector<IndexT>& shape)
```

Comment:
Why does this need a different type for `RetTypeT`? Shouldnt it just always be size_t?